Userland/Libraries/LibGfx/ImageFormats/JPEGWriter.cpp

```diff
@@ -135,6 +141,49 @@ class JPEGEncodingContext {
         return {};
     }
 
+    ErrorOr<void> initialize_mcu(CMYKBitmap const& bitmap)
+    {
+        u64 const horizontal_macroblocks = ceil_div(bitmap.size().width(), 8);
+        u64 const vertical_macroblocks = ceil_div(bitmap.size().height(), 8);
+        TRY(m_macroblocks.try_resize(horizontal_macroblocks * vertical_macroblocks));
+
+        for (u16 y {}; y < bitmap.size().height(); ++y) {
+            u16 const vertical_macroblock_index = y / 8;
+            u16 const vertical_pixel_offset = y - vertical_macroblock_index * 8;
+
+            for (u16 x {}; x < bitmap.size().width(); ++x) {
+                u16 const horizontal_macroblock_index = x / 8;
+                u16 const horizontal_pixel_offset = x - horizontal_macroblock_index * 8;
+
+                auto& macroblock = m_macroblocks[vertical_macroblock_index * horizontal_macroblocks + horizontal_macroblock_index];
+                auto const pixel_offset = vertical_pixel_offset * 8 + horizontal_pixel_offset;
+
+                auto const original_pixel = bitmap.scanline(y)[x];
+
+                // r is 255 - c (and similar for g/m b/y), but with the Adobe YCCK color transform marker, the CMY
+                // channels are stored inverted, which cancels out: 255 - (255 - x) == x.
+                u8 r = original_pixel.c;
+                u8 g = original_pixel.m;
+                u8 b = original_pixel.y;
```

Comment:
Calling these things `r`, `g`, `b` is quite confusing IMO.
The only thing this has in common with RGB is to serve as an input for the YCbCr transformation, right?

Reply:
It is confusing! The mathy, low-quality formula is `u8 r = 255 - c`. With this, you have a real `r`. But then adobe inverts its cmyk colors, so you have `u8 r = 255 - (255 - c);` which is the same as `u8 r = c;`. So I think c is actually == r, since it's stored inverted. I hope that makes sense (and I hope it's correct).

(If we one day want to store these CMYK jpegs in a pdf, we'll have to invert the channels once.)

Reply:
I tried tweaking the comment a bit. LMK if this is acceptable now :)

Reply:
Sure, let's go with that!